xdsl/dialects/riscv.py

```diff
@@ -152,6 +154,40 @@ def print_parameter(self, printer: Printer) -> None:
         printer.print_string(name)
 
 
+@irdl_attr_definition
+class SImm12Attr(IntegerAttr[IntegerType]):
+    """
+    A 12-bit immediate signed value.
+    """
+
+    name = "riscv.simm12"
+
+    def __init__(self, value: int) -> None:
+        super().__init__(value, IntegerType(12, Signedness.SIGNED))
+
+    # Pyright is confused by the parent class __new__ signature
+    def __new__(cls, value: int) -> SImm12Attr:
+        return super().__new__(cls)  # type: ignore
```

Comment:
type: ignore is almost never the right thing to use, I'm thinking of disabling it for the project. Could you please replace it with the specific pyright error?

Reply:
@math-fehr, do you know why it would be unhappy about __new__ here? Is there a recommended way to subclass Attributes?